tests/phpunit/tests/wp-token-map/wpTokenMap.php

```diff
@@ -0,0 +1,404 @@
+<?php
+/**
+ * Unit tests covering WP_Token_Map functionality.
+ *
+ * @package WordPress
+ *
+ * @since 6.6.0
+ *
+ * @coversDefaultClass WP_Token_map
+ */
+class Tests_WpTokenMap extends WP_UnitTestCase {
+	/**
+	 * Number of unique HTML5 named character references, including
+	 * variations of a given name that don't require the trailing semicolon.
+	 *
+	 * The set of names is fixed by the specification,
+	 * and can be found at the following link.
+	 *
+	 * @link https://html.spec.whatwg.org/entities.json
+	 */
+	const KNOWN_COUNT_OF_ALL_HTML5_NAMED_CHARACTER_REFERENCES = 2231;
+
+	/**
+	 * Small test array matching names to Emoji.
+	 *
+	 * @var array.
+	 */
+	const ANIMAL_EMOJI = array(
+		'cat'     => '🐈',
+		'dog'     => '🐶',
+		'fish'    => '🐟',
+		'mammoth' => '🦣',
+		'seal'    => '🦭',
+	);
+
+	/**
+	 * Returns an associative array whose keys are tokens to replace and
+	 * whose values are the replacement strings for those tokens.
+	 *
+	 * This function is here to help avoid bloating this specific test file.
+	 * For example, the HTML5 dataset is very large and best served as a
+	 * separate file.
+	 *
+	 * The HTML5 named character reference list is pulled directly from the
+	 * WHATWG spec and stored in the tests directory so it doesn't need to
+	 * be downloaded on every test run. By specification, it cannot change
+	 * and will not be updated.
+	 *
+	 * @ticket 60698
+	 *
+	 * @param string $dataset_name Which dataset to return.
+	 * @return array The dataset as an associative array.
+	 */
+	private static function get_test_input_array( $dataset_name ) {
+		static $html5_character_references = null;
+
+		switch ( $dataset_name ) {
+			case 'ANIMALS':
+				return self::ANIMAL_EMOJI;
+
+			case 'HTML5':
+				if ( ! isset( $html5_character_references ) ) {
+					$dataset = wp_json_file_decode(
+						__DIR__ . '/../../data/html5-entities/entities.json',
+						array( 'associative' => true )
+					);
+
+					$html5_character_references = array();
+					foreach ( $dataset as $name => $value ) {
+						$html5_character_references[ $name ] = $value['characters'];
+					}
+				}
+
+				return $html5_character_references;
+		}
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array[].
+	 */
+	public static function data_input_arrays() {
+		$dataset_names = array(
+			'ANIMALS',
+			'HTML5',
+		);
+
+		$datasets = array();
+
+		foreach ( $dataset_names as $dataset_name ) {
+			$datasets[ $dataset_name ] = array( self::get_test_input_array( $dataset_name ) );
+		}
+
+		return $datasets;
```

Comment:
This is another place where yielding is appropriate:

```suggestion
		foreach ( $dataset_names as $dataset_name ) {
			yield $dataset_name => array( self::get_test_input_array( $dataset_name ) );
		}
```